grakn-graph/src/main/java/ai/grakn/graph/internal/RelationTypeImpl.java

```diff
@@ -95,14 +98,34 @@ public RelationType deleteHasRole(RoleType roleType) {
 
         RoleTypeImpl roleTypeImpl = (RoleTypeImpl) roleType;
         //Add castings of roleType to make sure relations are still valid
-        roleTypeImpl.castings().forEach(casting -> getGraknGraph().getConceptLog().putConcept(casting));
+        roleTypeImpl.castings().forEach(casting -> getGraknGraph().getConceptLog().trackConceptForValidation(casting));
 
         //Add the Role Type itself
-        getGraknGraph().getConceptLog().putConcept(roleTypeImpl);
+        getGraknGraph().getConceptLog().trackConceptForValidation(roleTypeImpl);
 
         //Add the Relation Type
-        getGraknGraph().getConceptLog().putConcept(roleTypeImpl);
+        getGraknGraph().getConceptLog().trackConceptForValidation(roleTypeImpl);
+
+        //Remove from internal cache
+        if(cachedHasRoles.isPresent()) cachedHasRoles.get().remove(roleType);
+
+        //Remove from roleTypeCache
+        ((RoleTypeImpl) roleType).deleteCachedRelationType(this);
 
         return this;
     }
+
+    @Override
+    public void innerDelete(){
+        //Force load the cache
+        cachedHasRoles.get();
```

Comment:
yes because if the cache is not loaded then you will try to read from the database and traverse across the vertex AFTER it has been deleted. This will result in a good old null pointer.